modules/sdk-coin-sui/src/lib/iface.ts

```diff
@@ -14,7 +21,63 @@ export type SuiObjectRef = {
   digest: string;
 };
 
-export type TxDetails = PayTxDetails | PaySuiTxDetails | PayAllSuiTxDetails;
+export const ObjectId = string();
+// eslint-disable-next-line no-redeclare
+export type ObjectId = Infer<typeof ObjectId>;
+
+export const SuiAddress = string();
+// eslint-disable-next-line no-redeclare
+export type SuiAddress = Infer<typeof SuiAddress>;
+
```

Comment:
Naming it same for the variable and type will make it really confusing to utilise. I suggest to rename variables as `objectId` and `suiAddress` . Also any particular reason to use superstruct? I didn't see it being utilzed much in this coin package unless I missed something.

Reply:
nvm, I dropped this library, just wanted to align with the same types sui.js is using

Reply:
this can be useful cause superstruct checks the types of variables and inputs at runtime